.github/workflows/main.yml

```diff
@@ -0,0 +1,61 @@
+name: Ruby
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Ruby ${{ matrix.ruby }}
+    strategy:
+      matrix:
+        ruby: ['2.7', '3.0', '3.1', '3.2']
```

Comment:
Just a general note - I think we should drop support for 2.7 in line with Ruby EOL.

```
Ends in 2 months and 2 weeks(31 Mar 2023)
```